packages/synthetics-sdk-api/src/auto_instrumentation.ts

```diff
@@ -0,0 +1,179 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { getNodeAutoInstrumentations } from '@opentelemetry/auto-instrumentations-node';
+import { NodeTracerProvider } from '@opentelemetry/sdk-trace-node';
+import { registerInstrumentations } from '@opentelemetry/instrumentation';
+import {
+  BatchSpanProcessor,
+  AlwaysOnSampler,
+} from '@opentelemetry/sdk-trace-base';
+import { Span, TraceFlags } from '@opentelemetry/api';
+import { GoogleAuth, GoogleAuthOptions } from 'google-auth-library';
+import { Logger } from 'winston';
+import { TraceExporter } from '@google-cloud/opentelemetry-cloud-trace-exporter';
+
+const LOGGING_TRACE_KEY = 'logging.googleapis.com/trace';
+const LOGGING_SPAN_KEY = 'logging.googleapis.com/spanId';
+const LOGGING_SAMPLED_KEY = 'logging.googleapis.com/trace_sampled';
+const LOGGING_SEVERITY_KEY = 'severity';
+
+const levelToSeverityMap: { [key: string]: string } = {
+  error: 'ERROR',
+  warn: 'WARNING',
+  info: 'INFO',
+  http: 'INFO',
+  verbose: 'DEBUG',
+  debug: 'DEBUG',
+  silly: 'DEBUG',
+};
+
+let singletonAutoInstrumentation: SyntheticsAutoInstrumentation | null;
+
+/**
+ * @public
+ *
+ * This function sets up user authored synthetic code with a baseline open
+ * telemetry setup that will write traces and logs to cloud trace and cloud
+ * logging.
+ *
+ * NOTE: For this module to be used effectively, it needs to be included
+ * and ran before any other code within your synthetic application runs.
+ */
+export const instantiateAutoInstrumentation = (
+  args: { googleAuthOptions?: GoogleAuthOptions } = {}
+) => {
+  singletonAutoInstrumentation = new SyntheticsAutoInstrumentation(args);
+};
+
+/**
+ * @public
+ *
+ * Returns a winston logger that is instrumented to use the console transport.
+ *
+ * If {@link #instantiateAutoInstrumentation} is ran prior to any other code,
+ * and a project id is detected according to the logs will be instrumented
+ * with trace information that is formated in gcp's
+ * {@link https://cloud.google.com/logging/docs/structured-logging|structured logging}
+ * format.
+ */
+export const getInstrumentedLogger = async (): Promise<Logger> => {
+  if (singletonAutoInstrumentation) {
+    return await singletonAutoInstrumentation.getInstrumentedLogger();
+  } else {
+    const winston = require('winston');
+    return winston.createLogger({
+      transports: [new winston.transports.Console()],
+    });
+  }
+};
+
+class SyntheticsAutoInstrumentation {
+  provider: NodeTracerProvider;
+
+  private logger: Logger;
+  private gcpProjectId?: string | null;
+  private authArgs: GoogleAuthOptions;
+
+  constructor(args: { googleAuthOptions?: GoogleAuthOptions } = {}) {
+    this.authArgs = args.googleAuthOptions || {};
+
+    this.provider = new NodeTracerProvider({
+      sampler: new AlwaysOnSampler(),
+    });
+    const exporter = new TraceExporter();
```

Comment:
move this above or below the `this.***` lines

Reply:
ah i cant, its used in the next line "this.provider.addSpanProcessor(new BatchSpanProcessor(exporter))"